src/keygen/participant.rs

```diff
@@ -269,7 +262,8 @@ impl KeygenParticipant {
                 .iter()
                 .map(|msg| self.handle_round_two_msg(rng, msg, main_storage))
                 .collect::<Result<Vec<_>>>()?;
-            round_one_outcome.consolidate(round_two_outcomes)
+
+            Ok(ProcessOutcome::collect(round_two_outcomes)?.with_messages(round_one_messages))
```

Comment:
This pattern repeats a bunch and I wonder if it would be cleaner as a `ProcessOutcome` method, something like `collect_with_messages`?